app/Observers/DeviceObserver.php

```diff
@@ -164,6 +164,7 @@ public function deleting(Device $device): void
         $device->vrfLites()->delete();
         $device->vServers()->delete();
         $device->wirelessSensors()->delete();
+        \DB::table('applications')->where('device_id', $device->device_id)->delete();
```

Comment:
There is a line above that already deletes applications, just change that to:
`$device->applications()->forceDelete();`